src/Analyser/NodeScopeResolver.php

```diff
@@ -3928,6 +3930,49 @@ private function processClosureNode(
 		?Type $passedToType,
 	): ProcessClosureResult
 	{
+		if ($stmt instanceof Node\Stmt\Expression && $stmt->expr instanceof Expr\CallLike) {
```

Comment:
This needs to work for nested expressions to so to eliminate everything that isn't `Node\Stmt\Expression` is not ideal.

Reply:
We also need to do the same thing for `processArrowFunctionNode`.

Reply:
> This needs to work for nested expressions to so to eliminate everything that isn't Node\Stmt\Expression is not ideal.

How to do that?

Reply:
I think my other suggestion (to move to processArgs) will take care of this too.